### DISCUSSION TITLE
Reading selected columns from csv using fread

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/datatable#3418] [Q&A] [answered]
Hi!I am trying to read csv files into a datatable. I need to read only specific columns from the csv file. How do I do this for a datatable?Currently, I am reading the entire file into memory and then selecting only the required subset, but I would like to use a neater way for achieving this.

**Comment by oleksiyskononenko** [comment, marked as answer]
Just use the `columns` parameter in `fread()`: https://datatable.readthedocs.io/en/latest/api/dt/fread.html#columns

**Reply by n-srinidhi** [reply]
I used the following code:
`data=dt.fread(inpPath,columns=['Login','Profit','TimeCreate','TimeUpdate','Volume','PositionId','Symbol'])`
It gave me the following error:

```
Traceback (most recent call last):
  C:\Users\srini\Desktop\Projects\Centra\Web_App\src\get_insights.py:21 in <module>     getInsights()
  C:\Users\srini\Desktop\Projects\Centra\Web_App\src\get_insights.py:16 in getInsights  p_data=read_data('data/positions.csv')
  C:\Users\srini\Desktop\Projects\Centra\Web_App\src\get_insights.py:11 in read_data    data=dt.fread(inpPath,columns=['Login','Profit','TimeCreate','TimeUpdate','Volume','PositionId','Symbol'])
  $PY/datatable\utils\fread.py:379 in _override_columns                                 return _apply_columns_list(colspec, coldescs)
  $PY/datatable\utils\fread.py:439 in _apply_columns_list                               raise ValueError("Input contains %s, whereas `columns` "
  (with $PY = C:\Users\srini\Desktop\Projects\Centra\centra\lib\site-packages\)

ValueError: Input contains 40 columns, whereas columns parameter specifies only 7 columns
```

**Reply by n-srinidhi** [reply]
Ok. I changed the assignment to a set instead of a list and it worked. Thank you!
`data=dt.fread(inpPath,columns={'Login','Profit','TimeCreate','TimeUpdate','Volume','PositionId','Symbol'})`

**Reply by oleksiyskononenko** [reply]
Yeah, we need to adjust documentation, it is not really helpful with respect to the `columns` parameter: https://github.com/h2oai/datatable/issues/2847

**Reply by oleksiyskononenko** [reply]
Here are some examples of how it could be used: https://datatable.readthedocs.io/en/latest/manual/fread_examples.html

**Reply by n-srinidhi** [reply]
Thanks!
